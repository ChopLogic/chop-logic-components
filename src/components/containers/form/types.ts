export type ChopLogicFormData = { [key: string]: unknown };

export type ChopLogicFormValidationState = [string, boolean][];

export type ChopLogicFormInput = HTMLInputElement | HTMLSelectElement;

export type ChopLogicFormInputParams = { name: string; value: unknown; valid?: boolean };

export type ChopLogicFormContextProps = {
  onChangeFormInput?: (params: ChopLogicFormInputParams) => void;
  initialValues?: ChopLogicFormData;
  resetSignal?: number;
};
