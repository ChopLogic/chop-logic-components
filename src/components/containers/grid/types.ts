export type ChopLogicGridProps = React.HTMLAttributes<HTMLElement> & {
  columns: ChopLogicGridColumn[];
  data: ChopLogicGridItem[];
  caption?: string;
  id?: string;
  selectable?: boolean;
  renderDataItem?: (item: ChopLogicGridItem) => JSX.Element;
  onSelect?: (ids: string[]) => void;
};

export type ChopLogicGridColumn = {
  field?: string;
  title?: string;
  component?: React.ReactElement;
};

export type ChopLogicGridItem = {
  id: string;
  disabled?: boolean;
  [key: string]: unknown;
};
