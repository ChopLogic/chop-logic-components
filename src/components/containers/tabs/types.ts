import { ReactElement } from 'react';
import { CommonComponentProps } from 'utils/types';

export type ChopLogicTabsProps = CommonComponentProps & {
  tabs: ChopLogicTabItem[];
  defaultTabId?: string;
  mode?: ChopLogicTabsMode;
};

export type ChopLogicTabItem = {
  content: ReactElement;
  title: string;
  id: string;
  disabled?: boolean;
};

export type ChopLogicTabsMode = 'vertical' | 'horizontal';
