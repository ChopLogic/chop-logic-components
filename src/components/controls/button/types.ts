import { MouseEventHandler } from 'react';
import { Icon } from 'enums/icon';

export type ChopLogicButtonView = 'primary' | 'secondary' | 'danger' | 'icon';

export type ChopLogicButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
  text?: string;
  onClick?: MouseEventHandler<HTMLButtonElement> | (() => void);
  view?: ChopLogicButtonView;
  disabled?: boolean;
  icon?: Icon;
  label?: string;
};
