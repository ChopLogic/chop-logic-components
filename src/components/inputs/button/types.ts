import { MouseEventHandler } from 'react';
import { CommonInputProps } from 'utils/types';

import { ChopLogicIconName } from 'components/misc/icon/Icon';

export type ChopLogicButtonView = 'primary' | 'secondary' | 'danger' | 'icon';

export type ChopLogicButtonProps = Partial<CommonInputProps> & {
  onClick?: MouseEventHandler<HTMLButtonElement> | (() => void);
  view?: ChopLogicButtonView;
  icon?: ChopLogicIconName;
  text?: string;
  extended?: boolean;
  type?: 'button' | 'reset' | 'submit';
};
