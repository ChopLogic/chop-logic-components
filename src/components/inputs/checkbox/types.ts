export type ChopLogicCheckboxProps = React.InputHTMLAttributes<HTMLInputElement> & {
  id: string;
  name: string;
  label: string;
  isLabelHidden?: boolean;
};
