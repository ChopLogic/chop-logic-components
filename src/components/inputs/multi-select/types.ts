export type MultiSelectValue = {
  id: string;
  label: string;
  selected: boolean;
};

export type ChopLogicMultiSelectProps = React.SelectHTMLAttributes<HTMLSelectElement> & {
  id: string;
  name: string;
  label: string;
  values: MultiSelectValue[];
  onSelectChange?: (values?: MultiSelectValue[]) => void;
  placeholder?: string;
};

export type MultiSelectDropdownProps = {
  values: MultiSelectValue[];
  isOpened: boolean;
  dropdownId: string;
  onClose: () => void;
  onSelect: (id: string) => void;
};
