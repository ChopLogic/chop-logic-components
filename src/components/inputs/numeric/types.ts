export type NumericValidationFunction = (input?: number) => boolean;

export type ChopLogicNumericInputProps = React.InputHTMLAttributes<HTMLInputElement> & {
  name: string;
  label: string;
  errorMessage?: string;
  validator?: NumericValidationFunction;
  onClear?: () => void;
  hasSpinButtons?: boolean;
  onSpinButtonClick?: (value?: number) => void;
};
