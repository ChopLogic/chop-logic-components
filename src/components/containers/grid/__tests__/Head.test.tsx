import { render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';

import ChopLogicGridHead from '../elements/Head';
import { GridColumn } from '../types';

describe('ChopLogicGridHead', () => {
  const gridColumns: GridColumn[] = [
    { title: 'Col 1', field: 'field1' },
    { title: 'Col 2', field: 'field2' },
    { title: 'Col 3', field: 'field3' },
  ];

  const testProps = {
    columns: gridColumns,
    gridId: 'test-grid-id',
    selectable: true,
    selectAll: vi.fn(),
    deselectAll: vi.fn(),
  };

  it('should match the snapshot', () => {
    const { asFragment } = render(<ChopLogicGridHead {...testProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
