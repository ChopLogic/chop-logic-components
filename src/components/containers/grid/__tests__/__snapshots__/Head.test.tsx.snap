// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChopLogicGridHead > should match the snapshot 1`] = `
<DocumentFragment>
  <table>
    <thead
      class="_header_4a2c43"
    >
      <tr
        class="_header_row_4a2c43"
      >
        <th
          class="_header_cell_4a2c43"
        >
          <div
            class="_checkbox_4a2c43"
          >
            <input
              class="_checkbox_input_4a2c43"
              id="select_all_test-grid-id"
              type="checkbox"
            />
            <label
              class="_label_87260b"
              for="select_all_test-grid-id"
            >
              <svg
                fill="currentColor"
                height="1em"
                role="img"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M14 0H2C.9 0 0 .9 0 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V2c0-1.1-.9-2-2-2zm0 14H2V2h12v12z"
                  fill="currentColor"
                />
              </svg>
            </label>
          </div>
        </th>
        <th
          class="_header_cell_4a2c43"
        >
          Col 1
        </th>
        <th
          class="_header_cell_4a2c43"
        >
          Col 2
        </th>
        <th
          class="_header_cell_4a2c43"
        >
          Col 3
        </th>
      </tr>
    </thead>
  </table>
</DocumentFragment>
`;
